### PR TITLE
Introduced tabs/pages, single page view

### LINKED ISSUE #97: Implement tabs/pages
tabs/pages:

- featured
- recent
- favs
- issues

design: https://github.com/mozilla/network-pulse/issues/59

### PR DESCRIPTION
**Pull request by mmmavis**
Fixes #97, #107 

This is a BIGGGG PR. I wanted to keep it small but sometimes it's just hard to implement something without modifying/rewriting existing code. We can use the [QA checklist](https://github.com/mozilla/network-pulse/wiki/Quality-Tests) to make sure things work as expected.

Known bug / things to be implemented:

- The "issues" tab view will be implemented in another PR. This will also involve changes to the data formatter.
- Search functionality: currently it searches within the current tab (vs all projects). I'll file a separate PR to fix it along with updates mentioned in https://github.com/mozilla/network-pulse/issues/99.

/cc @acabunoc @cadecairos let me know if I can help clarify anything! 😃

### CODE REVIEW

**Comment by mmmavis** [comment]
### Splash screen
 - [ ] shows briefly while app loads

### Default view
 - [ ] cards are sorted by most recent at the top
 - [ ] cards display in columns to fit browser width responsively
 - [ ] cards contain title, a paragraph or two of text, two buttons, two icons
 - [ ] `coming soon` height of card fits content 

### Item card
 - [ ] Visit button opens a external link in a new tab
 - [ ] Get Involved button opens a external link in a new tab
 - [ ] link button reveals a URL to share (selected to copy on desktop)
 - [ ] Fav icon shows selected state when tapped

### Item shared. User opens link.
 - [ ] single item detail view displayed
 - [ ] search nav bar is showing with no menu items selected
 - [ ] no weird scrolling in the background

### Search
 When a user begins typing:
 - [ ] cancel search icon appears
 - [ ] hint text is hidden
 - [ ] only applicable cards are visible
 - [ ] `coming soon` nav (and tags) are hidden
 - [ ] `coming soon` add item button is hidden
 - [ ] `coming soon` search with zero results shows message

### Add Post Icon
 - [ ] Opens form
 - [ ] Add icon turns to Close icon
 - [ ] Form scrolls as expected
 - [ ] post is added to the spreadsheet
 - [ ] card is added to the site

### Footer
 - [ ] sign up opens google form in a tab
 - [ ] repo link opens in a tab

### Featured view `coming soon`
 - [ ] this is default view for a new visitor
 - [ ] only featured cards load
 - [ ] cards are sorted by most recent at the top

### Recent View `coming soon`
 - [ ] all cards load
 - [ ] cards are sorted with most recent at the top

### Favs View `coming soon`
 - [ ] shows message if no items have been faved
 - [ ] shows all faved cards
 - [ ] most recently favourited project should appear first in the list
 - [ ] fav icon shows selected state
 - [ ] clicking on the star again un-favs it and removes from this screen

### Categories/Issues screen `coming soon`
 - [ ] on landing users see message about issues
 - [ ] tapping on an issue only shows cards related to that issue
 - [ ] cards are sorted by most recent at the top
 - [ ] only one tag can be selected at a time

### 404 screen `coming soon`
 - [ ] shows error message for /brknURLs
 - [ ] links to repo

File: js/detail-view-manager.js [hunk 1]
```diff
@@ -1,47 +0,0 @@
-/* Detailer View Manager */
```
Comment (mmmavis) [inline js/detail-view-manager.js]: removing lightbox - this file is no longer needed

File: js/pulse.js [hunk 2]
```diff
@@ -171,28 +151,32 @@ var PulseMaker = {
...
+    // routes handler
+    ViewsManager.init(sortedProjects);
+
     if (FEATURE.notify) { 
       var updated = Notifier.checkForUpdates(sortedProjects);
       console.log('updated: ',updated);
     }
     if (updated || firstRun) {
+      // TODO:FIXME: is this check needed?
+      // else block means data update was triggered by refresh
```
Comment (mmmavis) [inline js/pulse.js]: just a TODO reminder for me

File: js/pulse.js [hunk 3]
```diff
@@ -171,28 +151,32 @@ var PulseMaker = {
...
     PulseMaker.toggleAdditionalFeatures();
     setTimeout(function(){ 
       PulseMaker.dismissSplash();
     }, 250);
-
-    // detail view handler
-    DetailViewManager.init();
   },
   'dismissSplash' : function() {
-    var siteHeader = document.getElementById('site-header');
```
Comment (mmmavis) [inline js/pulse.js]: just renaming `siteHeader` to `splashScreen`

File: js/pulse.js [hunk 4]
```diff
@@ -203,26 +187,15 @@ var PulseMaker = {
...
-    });
-  },
   'refresh' : function () {
     console.log('refresh');
     PulseMaker.getData(false);
   },
   'clearProjectLists' : function () {
-    fadeUpdate.fadeOut();
     document.getElementById('loading').style.display = 'block';
-    var lists = document.querySelectorAll('.project-list');
```
Comment (mmmavis) [inline js/pulse.js]: replaced by `renderProjectsIntoView` in `views-manager.js`

File: js/views-manager.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      $("#fav-not-found").hide();
+    } else {
+      $("#fav-not-found").show();
+    }
+    
+    $("#favs-view-link").addClass("active");
+    // TODO:FIXME: update url
+    window.history.pushState("Network Pulse", "", window.location.href.split("?")[0]);
+  },
+  showIssuesView: function() {
```
Comment (mmmavis) [inline js/views-manager.js]: will implement the issue tab view in upcoming PR

File: js/views-manager.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    // TODO:FIXME: issue view hasn't been not implemented yet
+    this.resetView();
+    // this.renderAllProjectsIntoView();
+    // this.$controlsContainer.show();
+    // $(".issue-btn").on("click",IssueBtnClickHandler);
+    this.$projectsContainer.append("<h3 style='text-align: center; width: 100%; margin-bottom: 200px;'>Coming soon...</h3>");
```
Comment (mmmavis) [inline js/views-manager.js]: temporary "coming soon" message

**Comment by cadecairos** [comment]
I'm seeing missing padding when I make my browser window smaller (although padding is present in responsive design mode)

screenshot from 2016-08-05 16-04-36

**Comment by cadecairos** [comment]
Not sure if desired or not, but when I visit a direct link for a project item, the splash screen appears under the lightbox (or at least before the lightbox fades out.)

**Comment by cadecairos** [comment]
Could we make it so the fav icon doesn't get selected when you click on it? (Firefox)

**Comment by cadecairos** [comment]
Should the exit search button and add item button overlap like this?

screenshot from 2016-08-05 16-36-03

**Comment by mmmavis** [comment]
@cade the overlap issue will be fixed in https://github.com/mozilla/network-pulse/issues/99

**Comment by cadecairos** [comment]
> repo link opens in a tab

This doesn't open in a new tab

**Comment by cade** [comment]
Thank goodness! That overlap was driving me crazy!!

609299a1128b20719e1ce667a0b10bd8bd11267167e1ab5fbe3af6fb74cd30f9

**Comment by mmmavis** [comment]
@cade omg i thought i've been pretty at tagging @cadecairos ... until now. Sorry!

**Comment by ScottDowne** [comment]
The real cade strikes again.

**Comment by cade** [comment]
I thought you were just missing me...

**Comment by mmmavis** [comment]
we just love to always keep you informed! 👍 😁

**Comment by cadecairos** [comment]
> Add icon turns to Close icon

This doesn't appear to happen

**Comment by mmmavis** [comment]
> Add icon turns to Close icon

@cadecairos sorry forgot to tag this with `coming soon`

File: js/views-manager.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
+// TODO:FIXME: 
+// search result layout
+// auto refresh?
+// issues view: coming soon
+
+var IssueBtnClickHandler = function(event) {
+  var issue = $(this).data("issue");
+  console.log("issue = ", issue);
```
Comment (cadecairos) [inline js/views-manager.js]: Let's drop the `console.log`

File: js/views-manager.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    $("#fav-not-found").hide();
+    this.$controlsContainer.hide();
+    this.$projectsContainer.children().remove();
+    $(".nav-item").removeClass("active");
+  },
+  init: function(allProjects) {
+    this.projects = allProjects;
+    var projectId = utility.getProjectIdFromUrl(window.location.href);
+    if ( projectId ) {
+      console.log(projectId);
```
Comment (cadecairos) [inline js/views-manager.js]: Let's drop the `console.log`

File: js/views-manager.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+  renderAllProjectsIntoView: function() {
+    this.renderProjectsIntoView(this.projects);
+  },
+  showSingleProjectView: function(projectId) {
+    this.resetView();
+    this.renderAllProjectsIntoView();
+    $(".project").hide();
+    $(".project[data-id="+projectId+"]").addClass("single").show();
+    // FIXME: just a quick solution to add id query param to URL
+    window.history.pushState(projectId, "", window.location.href.split("?")[0] + "?id=" + projectId);
```
Comment (cadecairos) [inline js/views-manager.js]: From what I can tell the only code path that gets to this function has already parsed the id out of the url, am I missing something?

Comment (mmmavis) [inline js/views-manager.js]: not sure if i understand ^ correctly @cadecairos but i think this `pushState` line can be removed.

**Comment by mmmavis** [comment]
:shipit: 🎉

**Pull request merged**